src/ZoneServer/Database/ZoneDb.cs

```diff
@@ -1443,5 +1446,100 @@ private void LoadCollections(Character character)
 				character.Properties.InvalidateAll();
 			}
 		}
+
+		/// <summary>
+		/// Inserts companion in database.
+		/// </summary>
+		/// <param name="accountId"></param>
+		/// <param name="characterId"></param>
+		/// <param name="companion"></param>
+		/// <returns></returns>
+		public void CreateCompanion(long accountId, long characterId, Companion companion)
+		{
+			using (var conn = this.GetConnection())
+			using (var trans = conn.BeginTransaction())
+			{
+				using (var cmd = new InsertCommand("INSERT INTO `companions` {0}", conn, trans))
+				{
+					companion.AdoptTime = DateTime.Now;
+
+					cmd.Set("accountId", accountId);
+					cmd.Set("characterId", characterId);
+					cmd.Set("name", companion.Name);
+					cmd.Set("monsterId", companion.Id);
+					cmd.Set("stamina", companion.Stamina);
+					cmd.Set("exp", companion.Exp);
+					cmd.Set("adoptTime", companion.AdoptTime);
+					cmd.Set("active", companion.IsActivated ? 1 : 0);
+
+					cmd.Execute();
+					companion.DbId = cmd.LastId;
+				}
+
+				trans.Commit();
```

Comment:
While not an issue per se, you don't need a transaction if there's only one query.